base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -544,6 +551,7 @@ where D: Digest + Send + Sync
 
     // Construct a pruned mmr for the specified MMR tree based on the checkpoint state and new additions and deletions.
     fn get_pruned_mmr(&self, tree: &MmrTree) -> Result<PrunedMutableMmr<D>, ChainStorageError> {
+        let _ = self.lock_for_read()?;
```

Comment:
I think this will drop the lock guard immediately as the `lock_for_read` function returns. 

```suggestion
        let _lock = self.lock_for_read()?;
```

this will drop at the end of the `get_pruned_mmr` function. Same for a few other places

Reply:
You are correct